python/hail/matrixtable.py

```diff
@@ -2441,14 +2447,24 @@ def _process_joins(self, *exprs):
 
         all_uids = []
         left = self
-        used_uids = set()
+        used_joins = set()
```

Comment:
is this hard to abstract between MatrixTable and Table? the code looks identical beyond the `annotateGlobalJSON` call.